src/Serialization/GAP.jl

```diff
@@ -0,0 +1,249 @@
+##############################################################################
+#
+# General ideas of the (de)serialization of GAP objects:
+#
+# - We define (de)serialization methods for GAP objects depending on
+#   GAP filters, and store them via calls to `install_GAP_serialization`
+#   and `install_GAP_deserialization`, at compile time.
+#
+# - We install these methods at runtime (when the OscarInterface package
+#   of GAP gets loaded) as methods for the GAP operations `SerializeInOscar`
+#   and `DeserializeInOscar`.
+
+#############################################################################
+#
+# utilities for installing methods depending on GAP filters
+#
+const _GAP_serializations = Pair{Symbol, Function}[]
+const _GAP_deserializations = Pair{Symbol, Function}[]
+
+function install_GAP_serialization(filtsymbol::Symbol, meth::Function)
+  push!(_GAP_serializations, filtsymbol => meth)
+  return
+end
+
+function install_GAP_deserialization(filtsymbol::Symbol, meth::Function)
+  push!(_GAP_deserializations, filtsymbol => meth)
+  return
+end
+
+function _describe_GAP_flags(flags::GapObj)
+  return [string(GAPWrap.NameFunction(x))
+          for x in GAPWrap.ELMS_LIST(GAP.Globals.FILTERS,
+                     GAP.Globals.TRUES_FLAGS(flags))]
+#TODO: omit implied filters
+end
+
+function _GAP_flags(filtnames::Vector)
+  # We need `GAP.evalstr` because of function names involving brackets.
+  l = [GAP.Globals.FLAGS_FILTER(GAP.evalstr(x))::GapObj
+       for x in filtnames]
```

Comment:
I don't understand this code, and I could not find any callers to `_GAP_flags` so I am not sure how to test it. If `_GAP_flags` is removed this is of course perfectly fine.

Reply:
This belonged to experiments about setting certain filters programmatically, that is, to collect in the serialization  what GAP knows and to set this in the deserialization. I think this was a dangerous idea. Now I have removed this code.